src/overlays/ovl_i6/fox_ve2.c

```diff
@@ -0,0 +1,4132 @@
+                sp74.x = Rand_ZeroOne() * 50.0f + 120.0f;
+                sp74.y = 0.0f;
+                sp74.z = 0.0f;
+                Matrix_MultVec3fNoTranslate(gCalcMatrix, &sp74, &sp68);
+            }
+            Math_SmoothStepToF(&D_80177A48[5], 3.0f, 1.0f, 0.05f, 0.0f);
+            player->camEye.z += player->vel.z * D_80177A48[2];
+            if (D_80177A80 > 230) {
+                player->unk_08C -= 7.0f;
+            }
+            if (D_80177A80 > 230) {
+                Math_SmoothStepToF(&D_80177A48[7], 3.0f, 1.0f, 0.1f, 0.0f);
+                player->unk_034 -= D_80177A48[7];
+            }
+
+            for (i = 0; i < sp90; i++) {
+                func_8007797C(player->pos.x + sp68.x, player->pos.y + sp68.y,
+                              player->pos.z - (D_80177A48[3] + D_80177A48[4]), 0.0f, 0.0f, 50.0f,
+                              (Rand_ZeroOne() * 2.5f) + 2.5f);
+            }
+
+            Math_SmoothStepToF(&player->camEye.x, D_80177A48[1] + player->pos.x, 0.1f, 15.0f, 0.0f);
+            Math_SmoothStepToF(&player->camEye.y, player->pos.y + 30.0f, 0.1f, 15.0f, 0.0f);
+            Math_SmoothStepToF(&player->unk_0EC, -180.0f, 0.02f, D_80177A48[6], 0.0f);
+            Math_SmoothStepToF(&D_80177A48[6], 3.0f, 1.0f, 0.05f, 0.0f);
+            D_801779A0 = player->pos.x;
+            D_801779B8 = player->pos.y;
+            D_801779C0 = player->unk_138 + D_80177D20;
+            if (D_80177A80 > 200) {
+                player->unk_25C += 0.03f;
+                if (player->unk_25C > 0.6f) {
+                    player->unk_25C = 0.6f;
+                }
+                Math_SmoothStepToF(&player->unk_034, 3.0f, 0.5f, 0.1f, 0.0f);
+                Math_SmoothStepToF(D_801779A8, 70.0f, 1.0f, 2.0f, 0.0f);
+            }
+
+            switch (D_80177A80) {
+                case 60:
+                    for (i = 0; i < ARRAY_COUNT(gEffects); i++) {
+                        if ((gEffects[i].obj.id == OBJ_EFFECT_383) && (gEffects[i].obj.status == 2)) {
+                            Object_Kill(&gEffects[i].obj, &gEffects[i].sfxPos);
+                            break;
+                        }
+                    }
+                    break;
+                case 25:
+                    for (i = 0; i < 12; i++) {
+                        func_i6_80193668(&gObjects80[i], player->pos.x, player->pos.y,
+                                         (player->camEye.z - D_80177D20) + (2195.0f * i), i);
+                    }
+                    break;
+                case 150:
+                    Audio_PlaySfx(0x1900602A, &D_800C5D28, 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                    break;
+                case 340:
+                    func_8001A55C(&player->unk_460, 0x11403071);
+                    func_8001D400(0);
+                    D_800D2F68 = 0;
+                    gBlurAlpha = 0xFF;
+                    gCurrentLevel = LEVEL_VENOM_2;
+                    D_8017827C = LEVELMODE_ALL_RANGE;
```

Comment:
I don't think this should be using the levelmode enum. Looks like it's a boolean, so should use true/false.

Reply:
alright i think that was a mips2c thing? not sure if it was implying there was a chain assign or not :P